kart/checkout.py

```diff
@@ -135,9 +97,13 @@ def checkout(
             fetched_envelope
             and not resolved_spatial_filter_spec.is_within_envelope(fetched_envelope)
         )
+    else:
+        # We also allow switching of spatial filter by just writing it to the config and then running
+        # `kart checkout`. TODO this may not be a good idea.
```

Comment:
Good question. I guess since we'd rather the user used a command to change the spatial filter, since then we can a) refuse or warn the user if it's not a good time to change it, and b) so we can enact the change immediately (and prevent the WC get out of sync with the spatial filter).

But since we can't prevent the user from messing with the spatial filter in the config, I think it's good to handle this as best we can. Updated the documentation here.